setup.py

```diff
@@ -0,0 +1,149 @@
+                              version_file, re.M)
+    if version_match:
+        return version_match.group(1)
+    raise RuntimeError("Unable to find version string.")
+
+
+cmdclass = {}
+if SPHINX_AVAILABLE:
+
+    class MakeDocs(BuildDoc):
+        """Port of old `cylc-make-docs`. Removed in #2989.
+
+        This class extends the Sphinx command class for setuptools. With
+        a difference that it tries to mimic the behaviour of the previous
+        `cylc-make-docs`.
+
+        So first it will execute `make-commands.sh`, which builds the
+        commands help information, in the appendices.
+
+        Then, instead of calling one builder, this class will call the
+        builder fot he single HTML, and also the builder for the multiple
+        HTML documentation.
+
+        Finally, one more tweak in this class is to move the doctrees
+        folder (in the same level as the documentation) to within the
+        documentation folder, named `.doctrees`, as before with
+        `cylc-make-docs`.
+        """
+
+        def run(self):  # type: () -> None
+            try:
+                self.spawn(["./doc/src/custom/make-commands.sh"])
+            except DistutilsExecError as e:
+                self.warn("Failed to run make-commands.sh")
+                raise e
+            self.do_run("html", "built-sphinx")
+            self.do_run("singlehtml", "built-sphinx-single")
+
+        def do_run(self, builder: str, output_dir: str):
+            """
+            Args:
+                builder (str): name of the Sphinx builder
+                output_dir (str): directory to write the documentation produced
+            """
+            self.builder = builder
+            self.builder_target_dirs = [
+                (builder, join(self.build_dir, output_dir))]
+            super().run()
+            # move doctrees to $build_dir/.doctrees
+            correct_doctrees = join(self.builder_target_dirs[0][1],
+                                    ".doctrees")
+            rmtree(correct_doctrees, ignore_errors=True)
+            move(self.doctree_dir, correct_doctrees)
+
+    cmdclass["build_sphinx"] = MakeDocs
+
+
+install_requires = [
+    'colorama==0.4.1',
+    'isodatetime==1!2.0.0',
+    'jinja2==2.10',
+    'markupsafe==1.1.1',
+    'python-jose==3.0.1',
+    'pyzmq==18.0.1'
+]
+tests_require = [
+    'codecov==2.0.15',
+    'coverage==4.5.3',
+    'pytest-cov==2.6.1',
+    'pytest==4.4.0',
+    'pycodestyle==2.5.0',
+    'testfixtures==6.6.2'
+]
+
+extra_requires = {
+    'empy': ['EmPy==3.3.4'],
+    'docs': ['sphinx==2.0.0'],
+    'all': []
+}
+extra_requires['all'] += extra_requires['empy']
+extra_requires['all'] += extra_requires['docs']
+extra_requires['all'] += tests_require
+
+setup(
+    version=find_version("lib", "cylc", "__init__.py"),
+    long_description=open('README.md').read(),
+    long_description_content_type="text/markdown",
+    scripts=glob(join('bin', '*')),
```

Comment:
This is probably me not understanding setuptools but do we need to remove the `cylc-` prefix here?

Reply:
>remove the cylc- prefix here?

Because of `scripts=glob(join('bin', '*')), `? Not sure if I understood this one. But the `scripts` will find all those scripts, and put them in the `$PATH` (which for me is `./venv/bin/cylc....` for virtual envs).

Then the `./venv/bin/cylc` command does some shell wizardry and is able to understand `cylc search` must execute `cylc-search`. Not sure if that's what you asked? Sorry.

Reply:
Ah so it's just adding them to the `$PATH`, OK, thanks for explaining.